### ISSUE TITLE
How do I apply boundary conditions?

### ISSUE THREAD

**Issue by sidkamat** [ucns3d-team/UCNS3D#47]
Hi. I was wondering how do I apply boundary conditions. I know BCs have to be assigned at the time of grid creation but let's say I want a velocity-inlet of 10 m/s, how do I do that? I can apply the velocity-inlet BC during the grid generation process, but how do input the 10 m/s. Is it just the _u-vel_ in the UCNS3D.DAT file? If yes, what do I do when I have two inlet conditions?

**Comment by sidkamat** [comment]
What if I have a single inflow? I tried specifying the inflow velocity using in the u-vel in UCNS3D.DAT but that seems to apply to the entire grid and not just the inflow boundary

**Comment by TakisCFD** [comment]
This is correct since the initialization depends on the initial condition profile flag. I will suggest that you go through the read me file and parameters file instructions since this will save you time figuring out what you need to change

**Comment by SRkumar97** [comment]
Hello!
I am requiring some help with setting up a turbulent boundary layer profile at the inlet of a 3D cuboidal domain (flat plate boundary layer). My doubt is, the profile.F90 comprises several flow profiles e.g. 95 is set for the TGV case. But all of these profiles seem to be applicable to the entire inlet domain. I wish to prescribe the basic primitive variables (rho, U, V, W, P) as a function of the domain length. i.e. if x=0 is the inlet domain, I wish to specify all these 5 variables distinctly for every cell across the domain, from y=0 to y=ymax. This is in the interest of prescribing a turbulent boundary layer profile at the inlet. Prescribing the entire profile in the profile.F90 looks a bit tedious, as the number of such data lines will be very bulky for fine meshes. Ideally, I want the code to read an entire inlet.dat ASCII file and interpolate the values into the corresponding grid cells, only in the inlet/inflow domain. Need some guidance on this, thanks for reading!

**Comment by sidkamat** [comment]
@SRkumar97 I think you can apply a velocity profile using the log law or the 1/7th power law.

**Comment by TakisCFD** [comment]
For the inlet turbulent boundary profile creation, you will need to modify the function inflow. The function inflow has already the x,y,z coordinates available for you to use and define the profiles that you want.
This should suffice to ensure the correct boundary layer profile is inserted.

**Comment by SRkumar97** [comment]
Thanks a lot, I'll look into the inflow function for defining the profile!

**Comment by SRkumar97** [comment]
Hello!
I thoroughly went through the **flow_operations.f90** which has got the function inflow, with profiles defined for several problems. I see the the **POX()** arrays correspond to position co-ordinates for X direction while **POY()** and **POZ()** arrays correspond to Y and Z-directions. And each of these arrays is of size 3. The point that is unclear is, what do each of POX(1), POX(2) and POX(3) correspond to in the actual mesh? Do each of these 3 cover a set of X-co-ordinates?

**Comment by TakisCFD** [comment]
I suggest that you look at the supersonic outflow option which enforces what you are looking for. You can manipulate the outflow2d or outflow behavior accordingly.

**Comment by SRkumar97** [comment]
Thank you very much for your inputs, will check out these options.
